### ISSUE TITLE
Warnings "is missing in props validation" when upgrading from 2.5.2 to 2.6.x.

### ISSUE THREAD

**Issue by steida** [yannickcr/eslint-plugin-react#139]
I had an isssue after update.

27:39  warning  'router.getCurrentQuery' is missing in props validation for Login  react/prop-types
28:22  warning  'router.replaceWith' is missing in props validation for Login      react/prop-types
2.5.2 works. File https://gist.github.com/steida/f2b2e9d6df01576c3fee

**Comment by mathieumg** [comment]
Didn't you already have an issue about this? See https://github.com/yannickcr/eslint-plugin-react/issues/137#issuecomment-117846606, we even discussed it?

**Comment by steida** [comment]
I tried everything what was suggested.

**Comment by yannickcr** [comment]
@mathieumg I asked to create another issue since it was a different problem than #137

**Comment by mathieumg** [comment]
My bad, I missed that line. To quote myself from that thread, @steida : 

> That would be a React issue though, can I see what you are giving to the router prop?

**Comment by yannickcr** [comment]
Seems it is a function with some custom methods, I don't think we currently handle this case. I'll have a look.

**Comment by mathieumg** [comment]
That's what I was wondering, perhaps `PropTypes.instanceOf` would have been more appropriate.

**Comment by steida** [comment]
Ok, I updated gulp-eslint, and it works. Damn, gulp wrappers are more miss than hit.

**Comment by steida** [comment]
Ups, no it still does not work. If I use shape, this warning is logged:
```
Warning: Failed propType: Invalid prop `router` of type `function` supplied to `Login`, expected `object`. Check the render method of `LoginExposeRouter`.
```

So, func is must for React. And eslint-plugin-react needs shape. So this is still an issue to solve.

**Comment by mathieumg** [comment]
@steida Do you mind posting what exactly the value you are giving to `router` looks like? Thank you!

**Comment by tleunen** [comment]
Yes, you should specify a function.
I think the issue is because you are using something 2 levels deep `router.getCurrentQuery().nextPath`. Because of that, eslint-react wants you to specify `getCurrentQuery` as well.

**Comment by steida** [comment]
I found it.

This code sucks:

```js
redirectAfterLogin() {
  const nextPath = this.props.router.getCurrentQuery().nextPath;
  this.props.router.replaceWith(nextPath || 'home');
}
```

This code works:

```js
redirectAfterLogin() {
  const {router} = this.props;
  const nextPath = router.getCurrentQuery().nextPath;
  router.replaceWith(nextPath || 'home');
}
```

**Comment by mathieumg** [comment]
It works because you go around the problem, your prop is just not getting validated at all anymore. It would help to understand the problem better if you would provide us with an example of the value you give to the `router` prop, thank you.

**Comment by steida** [comment]
fyi `const router = this.props.router` isn't validated as well.

**Comment by steida** [comment]
You can check it here https://github.com/este/este/blob/master/src/client/components/exposerouter.react.js

**Comment by andolf** [comment]
I'm having this same issue, but with immutable objects using [seamless-immutable](https://github.com/rtfeldman/seamless-immutable). The prop types provided to a component are immutable objects, and whenever I use methods added for that particular library such as `asMutable()`, I get errors that it's not specified. I could go around it like suggested above, but that sort of defeats the purpose.

Is it possible to ignore this?
